flash/image/classification/adapters.py

```diff
@@ -0,0 +1,416 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import functools
+import inspect
+import random
+from collections import defaultdict
+from functools import partial
+from typing import Any, Callable, List, Optional, Type
+
+import torch
+from pytorch_lightning import LightningModule
+from pytorch_lightning.trainer.states import TrainerFn
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from torch.utils.data import DataLoader, Sampler
+
+from flash.core.adapter import Adapter, AdapterTask
+from flash.core.data.auto_dataset import BaseAutoDataset
+from flash.core.data.data_source import DefaultDataKeys
+from flash.core.model import Task
+from flash.core.registry import FlashRegistry
+from flash.core.utilities.imports import _LEARN2LEARN_AVAILABLE
+from flash.core.utilities.providers import _LEARN2LEARN
+from flash.core.utilities.url_error import catch_url_error
+
+if _LEARN2LEARN_AVAILABLE:
+    import learn2learn as l2l
+
+    class RemapLabels(l2l.data.transforms.TaskTransform):
```

Comment:
Could we always define this even if l2l is not available? That way it can be added to API reference

Reply:
Hi @ethanwharris, as this is [something which is static](https://github.com/PyTorchLightning/lightning-flash/blob/b33beb7d507e470a6350351b4fd166ac28b88e88/flash/image/classification/adapters.py#L163-L169) (the user cannot change this) and you'll likely use it with l2l only, having it in the API refs might be confusing

Reply:
Done !